aws/resource_aws_servicecatalog_portfolio_product_association.go

```diff
@@ -0,0 +1,143 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicecatalog"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"log"
+	"strings"
+	"time"
+)
+
+func resourceAwsServiceCatalogPortfolioProductAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsServiceCatalogPortfolioProductAssociationCreate,
+		Read:   resourceAwsServiceCatalogPortfolioProductAssociationRead,
+		Update: resourceAwsServiceCatalogPortfolioProductAssociationUpdate,
+		Delete: resourceAwsServiceCatalogPortfolioProductAssociationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"portfolio_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"product_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsServiceCatalogPortfolioProductAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	_, portfolioId, productId := resourceAwsServiceCatalogPortfolioProductAssociationRequiredParameters(d)
+	input := servicecatalog.AssociateProductWithPortfolioInput{
+		PortfolioId: aws.String(portfolioId),
+		ProductId:   aws.String(productId),
+	}
+	conn := meta.(*AWSClient).scconn
+	_, err := conn.AssociateProductWithPortfolio(&input)
+	if err != nil {
+		return fmt.Errorf("creating Service Catalog Product(%s)/Portfolio(%s) Association failed: %s",
+			productId, portfolioId, err.Error())
+	}
+	return resourceAwsServiceCatalogPortfolioProductAssociationRead(d, meta)
+}
+
+func resourceAwsServiceCatalogPortfolioProductAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	id, portfolioId, productId := resourceAwsServiceCatalogPortfolioProductAssociationRequiredParameters(d)
+	input := servicecatalog.ListPortfoliosForProductInput{
+		ProductId: aws.String(productId),
+	}
+	conn := meta.(*AWSClient).scconn
+	var pageToken = ""
+	isFound := false
+	for {
+		pageOfDetails, nextPageToken, err := resourceAwsServiceCatalogPortfolioProductAssociationListPortfoliosForProductPage(conn, input, &pageToken)
+		if err != nil {
+			return err
+		}
+		for _, portfolioDetail := range pageOfDetails {
+			if *portfolioDetail.Id == portfolioId {
```

Comment:
warp `portfolioDetail.Id` with `aws.StringValue()`

Reply:
done (check-in to come)